lib/Entity.rb

```diff
@@ -77,9 +77,7 @@ def options_string
   end
 
   def mutate(rate)
-    if option != nil then
-        @options.each{ |option| if rand() < rate then option.mutate end}
-    end
+    @options.each{ |option| if rand() < rate && option != nil then option.mutate end}
```

Comment:
`option.nil?` is a preferable way in Ruby.

Reply:
OK. Thanks.